apps/web/src/server/public-api/hono.ts

```diff
@@ -1,11 +1,54 @@
-import { OpenAPIHono } from "@hono/zod-openapi";
 import { swaggerUI } from "@hono/swagger-ui";
-import { handleError } from "./api-error";
+import { OpenAPIHono } from "@hono/zod-openapi";
 import { env } from "~/env";
+import { db } from "~/server/db";
+import { handleError } from "./api-error";
+import { getApiKeyFromToken, getTeamFromToken } from "./auth";
 
 export function getApp() {
   const app = new OpenAPIHono().basePath("/api");
 
+  // Add logging middleware
+  app.use("*", async (c, next) => {
+    const start = performance.now();
+
+    await next();
+
+    const end = performance.now();
+
+    const duration = Math.round(end - start);
+
+    try {
+      const team = await getTeamFromToken(c);
+      const apiKey = await getApiKeyFromToken(c);
+      const requestBody = await c.req.json().catch(() => null);
+      const responseBody = c.res.body
+        ? await c.res
+            .clone()
+            .json()
+            .catch(() => null)
+        : null;
+
+      const log = await db.apiLog.create({
```

Comment:
will this delay the api response? if yes we should do it in separate queue

Reply:
This operation will execute in the same request cycle but after the response has been sent to the user, and we are cloning the request to get the response and other details to log.

Here is the execution order: 
1. Request comes
2. Mark the start time
3. Response is sent to the user
4. Mark the end time
5. Clone the request to get the response data
6. Store the log

So, I don't think the queue is necessary

Reply:
But there is one problem with this approach, I don't see any other better approach.

Since, we are logging in the middleware, but the response to the request is returned from the inside logic, hence we cannot tap into the original request to get the response and the returned status code, we had to clone the original request to into another and access those data. But cloning makes the express thinks, its a 2 separate request being handled which causes the rate limiting to count it as 2 requests I believe. Have to verify that. 

If thats the case, we have to make the rate limit value x 2 to compensate this changes.